docs/proposal/20250224-rosa-networking.md

```diff
@@ -0,0 +1,221 @@
+    ```
+
+   `RosaNetwork` spec example with specified availability zones:
+    ```
+    kind: RosaNetwork
+    metadata:
+      name: rosa-network-01
+      namespace: default
+    spec:
+      availabilityZones:
+      - us-west-2a
+      - us-west-2d
+      region: us-west-2
+      cidrBlock: 10.0.0.0/16
+    ```
+
+1. A new reconciler for the new custom resource, implementing creation and deletion. The reconciler will be using an existing [CloudFormation template from ROSA CLI](https://github.com/openshift/rosa/blob/master/cmd/create/network/templates/rosa-quickstart-default-vpc/cloudformation.yaml) and will use [AWS CloudFormation API](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation) to [create](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#Client.CreateStack) and [delete](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#Client.DeleteStack) the AWS CloudFormation stack.
+
+   Outputs and resources created in the cloudformation stack will be tracked under `status` of the `RosaNetwork` type. In particular, the `status` will contain the list of public and private subnets and availability zones, grouped together by the availability zones.
+
+   Example:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     subnets:
+     - availabilityZone: us-west-2a
+       publicSubnet: subnet-0d9f28ba991b93514
+       privateSubnet: subnet-1d9f28ba992a83514
+     - availabilityZone: us-west-2b
+       publicSubnet: subnet-2d7f18c09f1b43512
+       privateSubnet: subnet-2d7f58c09f1b43512
+     - availabilityZone: us-west-2c
+       publicSubnet: subnet-1d7e19c0af1c4c57f
+       privateSubnet: subnet-7d7e19c0af1f4d57f
+   ```
+
+   All resources created in the cloudformation stack will be tracked under `status.resources` array:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     resources:
+       - resource: NATGateway1
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Eventual consistency check initiated
+       - resource: VPC
+         id: vpc-055edf3ebf27f18d6 
+         status: CREATE_COMPLETE
+         reason:
+       - resource: SecurityGroup
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Resource creation Initiated
+   ```
+   and will be reflecting the the values coming from [AWS CloudFormation API](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#Client.DescribeStackEvents) (`resource`, `id`, `status` and `reason`).
+
+   `status.conditions` of the `RosaNetwork` resource will be consistent with the CAPA conventions, example of a successful network stack creation:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-20T14:45:26Z"
+       status: "True"
+       type: RosaNetworkReady
```

Comment:
How will the `true` status be calculated for this? That all resources are in `CREATE_COMPLETE` from the AWS API, or something else?

Reply:
I'll also mention the changes that happened in upstream CAPI with regards to the definition of `Ready`[1]. In this context, I believe it should mean "The network resources can pass traffic for workloads".

I don't think anything in this proposal is counter to what I just said, but I wanted to point that out for any other reviewers.

We should also try to adhere to the recommendations made in that CAEP. Since CAPA's conditions have not yet been updated, conditions defined by this CRD may not match the past conventions, but IMO we should use the new conventions when adding new resources.

[1] https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#readiness-and-availability

Reply:
> How will the `true` status be calculated for this? That all resources are in `CREATE_COMPLETE` from the AWS API, or something else?

AWS API informs the user / consumer when the whole stack is `CREATE_COMPLETE`.